### ISSUE TITLE
gnrc pktbuf: replace parameter checks with assert 

### ISSUE THREAD

**Issue by OlegHahm** [RIOT-OS/RIOT#5031]
Almost every function in `gnrc_pktbuf_static.c` seems to check its parameters for valid values. I guess we can replace most of these checks with asserts and save some ROM.

**Comment by authmillenon** [comment]
This needs to be done very carefully. Sometimes a parameter might be "legally" in an invalid change, e.g. size == 0 if there is no payload for a packet.

**Comment by OlegHahm** [comment]
> Sometimes a parameter might be "legally" in an invalid change, 

Can you rephrase, please?

**Comment by authmillenon** [comment]
An assert lets the device effectively crash if there is an invalid value, while checking with an if and return an error code does not. There might be reasons why the parameters were wrong that might be totally feasible in the current program logic that use the `gnrc_pktbuf` API.

**Comment by authmillenon** [comment]
(you don't save ROM if you have to check your all your parameter values before each call to `gnrc_pktbuf_add()` e.g., just do it in the function and return an error state)

**Comment by OlegHahm** [comment]
May a null pointer also origin from a valid situation?

**Comment by authmillenon** [comment]
For `gnrc_pktbuf_release()` e.g. I would say, yes. But for anything else I rephrase

> This needs to be done very carefully.

with "this" I mean which parameter states could be a legal state though invalid.

**Comment by authmillenon** [comment]
And by "done" I mean "analyzed" :P

**Comment by Yonezawa-T2** [comment]
Another concern is security. We should review the code to ensure no invalid parameters are passed to pktbuf even if malicious packets are injected.
